codex/slots/proofs/backends/circomcompat.nim

```diff
@@ -32,40 +35,24 @@ type
     numSamples    : int     # number of samples per slot
     r1csPath      : string  # path to the r1cs file
     wasmPath      : string  # path to the wasm file
-    zKeyPath      : string  # path to the zkey file
+    zkeyPath      : string  # path to the zkey file
     backendCfg    : ptr CircomBn254Cfg
-
-  CircomG1* = G1
-  CircomG2* = G2
-
-  CircomProof*  = Proof
-  CircomKey*    = VerifyingKey
-  CircomInputs* = Inputs
+    vkp*          : ptr CircomKey
 
 proc release*(self: CircomCompat) =
-  ## Release the backend
+  ## Release the ctx
```

Comment:
Risking painting the bike shed: I understand a lot better what "backend" is than "ctx". Ctx reads like "context", but not sure why you'd want to call this a context. Plus there's still uses of the word "backend" here and there and it's not clear if they were just forgotten or if we're now making a distinction between ctx and backend.

Reply:
The way I think about it is - `this is the context for the backend` and from that standpoint it makes sense.

Reply:
OK fine, it's sort of a context object which you init, put stuff into, hand over to the backend to get the proofs, then release. In that case I think the weird part is that you're interacting with an object that's called `cfg` to get things done (`backendCfg`), when I guess the expectation for a config object is that it contains read-only data.

Reply:
but, nothing is being written to the `cfg`, it does in fact, only contain data.